sale_order_type/models/sale.py

```diff
@@ -125,6 +125,13 @@ def _prepare_invoice(self):
             res["sale_type_id"] = self.type_id.id
         return res
 
+    def copy(self, default=None):
+        if self.type_id and self.type_id.analytic_account_id:
+            default = dict(
+                default or {}, analytic_account_id=self.type_id.analytic_account_id.id,
+            )
+        return super().copy(default)
```

Comment:
You should read the type on the defaults, as you don't know if there's any override or default value set on the DB (and respect kw args):

```suggestion
        res = super().copy(default=default)
        sale_type = self.env["sale.order.type"].browse(res.get("type_id", False)) or self.type_id
        if sale_type.analytic_account_id:
            res["analytic_account_id"] = sale_type.analytic_account_id.id
        return res
```